runtime/src/prioritization_fee_cache.rs

```diff
@@ -278,71 +289,88 @@ impl PrioritizationFeeCache {
     /// Internal function is invoked by worker thread to update slot's minimum prioritization fee,
     /// Cache lock contends here.
     fn update_cache(
-        cache: Arc<RwLock<LruCache<Slot, Arc<Mutex<PrioritizationFee>>>>>,
+        cache: Arc<RwLock<LruCache<Slot, Arc<SlotPrioritizationFee>>>>,
         slot: &Slot,
+        bank_id: &BankId,
         transaction_fee: u64,
         writable_accounts: Arc<Vec<Pubkey>>,
         metrics: Arc<PrioritizationFeeCacheMetrics>,
     ) {
-        let (block_prioritization_fee, cache_lock_time) =
+        let (slot_prioritization_fee, cache_lock_time) =
             measure!(Self::get_prioritization_fee(cache, slot), "cache_lock_time");
 
-        let (mut block_prioritization_fee, entry_lock_time) =
-            measure!(block_prioritization_fee.lock().unwrap(), "entry_lock_time");
-
         let (_, entry_update_time) = measure!(
-            block_prioritization_fee.update(transaction_fee, &writable_accounts),
+            {
+                let mut block_prioritization_fee = slot_prioritization_fee
+                    .entry(*bank_id)
+                    .or_insert(PrioritizationFee::default());
+                block_prioritization_fee.update(transaction_fee, &writable_accounts)
+            },
             "entry_update_time"
         );
         metrics.accumulate_total_cache_lock_elapsed_us(cache_lock_time.as_us());
-        metrics.accumulate_total_entry_lock_elapsed_us(entry_lock_time.as_us());
         metrics.accumulate_total_entry_update_elapsed_us(entry_update_time.as_us());
     }
 
     fn finalize_slot(
-        cache: Arc<RwLock<LruCache<Slot, Arc<Mutex<PrioritizationFee>>>>>,
+        cache: Arc<RwLock<LruCache<Slot, Arc<SlotPrioritizationFee>>>>,
         slot: &Slot,
+        bank_id: &BankId,
         metrics: Arc<PrioritizationFeeCacheMetrics>,
     ) {
-        let (block_prioritization_fee, cache_lock_time) =
+        let (slot_prioritization_fee, cache_lock_time) =
             measure!(Self::get_prioritization_fee(cache, slot), "cache_lock_time");
 
-        let (mut block_prioritization_fee, entry_lock_time) =
-            measure!(block_prioritization_fee.lock().unwrap(), "entry_lock_time");
-
         // prune cache by evicting write account entry from prioritization fee if its fee is less
         // or equal to block's minimum transaction fee, because they are irrelevant in calculating
         // block minimum fee.
         let (_, slot_finalize_time) = measure!(
```

Comment:
nit: seems the returned Result is silently discarded after assigning it to `_`. how about `error!()`-ing, at least?

side-note: i have hidden desire for `--deny clippy::let_underscore_must_use` tree-wide... ;)

Reply:
👍🏼  added `error!(...)` should Result is_err()